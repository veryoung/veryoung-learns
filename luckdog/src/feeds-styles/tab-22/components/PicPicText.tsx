/** 左图（2图）右文
 * @Author: liquid
 * @Date:   2019年10月15日21:01:50
 *
 */

import React from 'react';
import { View, Image, Text, StyleSheet } from '@tencent/hippy-react-qb';
import FeedsTheme from '../../../framework/FeedsTheme';
import { vectorToArray } from './utils';
import FeedsProtect from '../../../mixins/FeedsProtect';
import { shouldComponentUpdate, countReRender } from '@tencent/luckbox-react-optimize';
import { strictExposeReporter } from '@/luckdog';
import { CommonProps } from '../../../entity';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  imageView: {
    height: 68,
    width: 71,
  },
  imageMain: {
    position: 'absolute',
    right: 0,
    bottom: 0,
    opacity: 0.5,
  },
  imageMainSize: {
    width: 40,
    height: 52.5,
  },
  imageBorder: {
    borderColors: FeedsTheme.SkinColor.D2,
    borderWidth: 0.5,
    borderStyle: 'solid',
    borderRadius: 2,
  },
  imageRight: {
    position: 'absolute',
    left: 0,
    top: 0,
    width: 51,
    height: 68,
  },
  imageTag: {
    backgroundColor: '#242424dd',
    borderRadius: 2,
    position: 'absolute',
    right: 0,
    bottom: 0,
    width: 50,
    alignItems: 'center',
  },
  imageTagText: {
    fontSize: 12,
    colors: FeedsTheme.SkinColor.A5,
    lineHeight: 20,
  },
  textView: {
    flexDirection: 'column',
    flexWrap: 'wrap',
    flex: 1,
    marginLeft: 12,
  },
  textMain: {
    fontSize: 16,
    colors: FeedsTheme.SkinColor.A1,
    lineHeight: 24,
    textAlign: 'justify',
  },
  tagSplit: {
    width: 0.5,
    height: 7,
    marginHorizontal: 6,
    marginTop: 4,
    backgroundColors: FeedsTheme.SkinColor.A4,
  },
  tegText: {
    fontSize: 12,
    colors: FeedsTheme.SkinColor.A3,
    lineHeight: 16,
    marginTop: 4,
  },
});

interface Props extends CommonProps {
  cardIndex: number;
  sBookId: string;
  vTag: any;
  style: any;
  sImage: string;
  sSubImage: string;
  sSubText: string;
  sMainText: string;
  openUrl: () => void;
}

@FeedsProtect.protect
export class PicPicText extends React.Component<Props> {
  public constructor(props) {
    super(props);
    this.shouldComponentUpdate = shouldComponentUpdate(this, 'TextPic');
  }

  public onLayout = (event, bookId) => {
    strictExposeReporter.addExpoItem({
      cardIndex: this.props.cardIndex,
      bookId,
      rect: event.layout,
    });
  };

  public render() {
    countReRender(this, 'TextPic');
    const { props } = this;
    const { itemBean = {}, sBookId } = this.props;
    const vTag = vectorToArray(props.vTag);
    return (
      <View
        style={[styles.container, this.props.style]}
        onClick={strictExposeReporter.triggerExpoCheck(props.openUrl)}
        onLayout={event => this.onLayout(event, sBookId)}
      >
        <View style={styles.imageView}>
          <View style={styles.imageMain}>
            <Image
              style={[styles.imageMainSize, styles.imageBorder]}
              source={{ uri: props.sImage }}
              resizeMode='cover'
              reportData={{ sourceFrom: itemBean.item_id }}
            />
          </View>

          <Image
            style={[styles.imageRight, styles.imageBorder]}
            source={{ uri: props.sSubImage }}
            resizeMode='cover'
            reportData={{ sourceFrom: itemBean.item_id }}
          />
          <View style={styles.imageTag}>
            <Text style={styles.imageTagText}>{props.sSubText}</Text>
          </View>
        </View>
        <View style={styles.textView}>
          <View>
            <Text style={styles.textMain} numberOfLines={2}>
              {props.sMainText}
            </Text>
          </View>
          <View style={{ flexDirection: 'row' }}>
            {vTag.map((item, idx) => (
              <View key={idx} style={{ flexDirection: 'row', alignItems: 'center' }}>
                {idx && <View style={styles.tagSplit} />}
                <Text style={styles.tegText} numberOfLines={1}>
                  {item}
                </Text>
              </View>
            ))}
          </View>
        </View>
      </View>
    );
  }
}
