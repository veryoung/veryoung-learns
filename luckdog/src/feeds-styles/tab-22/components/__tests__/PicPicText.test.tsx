import React from 'react';
import renderer from 'react-test-renderer';

import { PicPicText } from '../PicPicText';
import { commonProps } from '@test/mocks/props';

describe('<PicPicText/>', () => {
  describe('render testing', () => {
    it('render normal', () => {
      const props = {
        ...commonProps,
        cardIndex: 1,
        sBookId: '',
        vTag: [],
        style: {},
        sImage: '',
        sSubImage: '',
        sSubText: '',
        sMainText: '',
        openUrl: () => { /*  */ },
      };
      const tree = renderer.create(<PicPicText {...props} />);
      expect(tree).toMatchSnapshot();
    });
  });
});
