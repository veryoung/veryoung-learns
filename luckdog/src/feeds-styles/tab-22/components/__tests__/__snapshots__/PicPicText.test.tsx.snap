// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PicPicText/> render testing render normal 1`] = `
<div
  nativeName="View"
  onClick={[Function]}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Object {},
    ]
  }
>
  <div
    nativeName="View"
    style={
      Object {
        "height": 68,
        "width": 71,
      }
    }
  >
    <div
      nativeName="View"
      style={
        Object {
          "bottom": 0,
          "opacity": 0.5,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <img
        alt=""
        nativeName="Image"
        reportData={
          Object {
            "sourceFrom": undefined,
          }
        }
        resizeMode="cover"
        style={
          Object {
            "0": Object {
              "height": 52.5,
              "width": 40,
            },
            "1": Object {
              "borderColors": Array [
                "#e5e5e5",
                "#31363b",
                "#00000010",
                "#ffffff10",
              ],
              "borderRadius": 2,
              "borderStyle": "solid",
              "borderWidth": 0.5,
            },
          }
        }
      />
    </div>
    <img
      alt=""
      nativeName="Image"
      reportData={
        Object {
          "sourceFrom": undefined,
        }
      }
      resizeMode="cover"
      style={
        Object {
          "0": Object {
            "height": 68,
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": 51,
          },
          "1": Object {
            "borderColors": Array [
              "#e5e5e5",
              "#31363b",
              "#00000010",
              "#ffffff10",
            ],
            "borderRadius": 2,
            "borderStyle": "solid",
            "borderWidth": 0.5,
          },
        }
      }
    />
    <div
      nativeName="View"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#242424dd",
          "borderRadius": 2,
          "bottom": 0,
          "position": "absolute",
          "right": 0,
          "width": 50,
        }
      }
    >
      <p
        nativeName="Text"
        style={
          Object {
            "colors": Array [
              "white",
              "#7f8386",
              "#ffffff20",
              "white",
            ],
            "fontSize": 12,
            "lineHeight": 20,
          }
        }
        text=""
      >
        
      </p>
    </div>
  </div>
  <div
    nativeName="View"
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "flexWrap": "wrap",
        "marginLeft": 12,
      }
    }
  >
    <div
      nativeName="View"
      style={Object {}}
    >
      <p
        nativeName="Text"
        numberOfLines={2}
        style={
          Object {
            "colors": Array [
              "#242424",
              "#686d74",
              "#242424",
              "white",
            ],
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "justify",
          }
        }
        text=""
      >
        
      </p>
    </div>
    <div
      nativeName="View"
      style={
        Object {
          "flexDirection": "row",
        }
      }
    />
  </div>
</div>
`;
